fl4health/clients/basic_client.py

```diff
@@ -387,14 +388,32 @@ def _handle_reporting(
         reporting_dict.update(metric_dict)
         self.wandb_reporter.report_metrics(reporting_dict)
 
-    def train_step(self, input: torch.Tensor, target: torch.Tensor) -> Tuple[TrainingLosses, Dict[str, torch.Tensor]]:
+    def _move_data_to_device(self, data: TorchInputType) -> TorchInputType:
+        if isinstance(data, torch.Tensor):
+            return data.to(self.device)
+        elif isinstance(data, dict):
+            return {key: value.to(self.device) for key, value in data.items()}
+        else:
+            raise TypeError("data must be of type Tensor or Dict[str, Tensor].")
+
+    def is_empty_batch(self, input: Union[torch.Tensor, Dict[str, torch.Tensor]]) -> bool:
+        if isinstance(input, torch.Tensor):
+            return len(input) == 0
+        elif isinstance(input, dict):
+            return all(len(val) == 0 for _, val in input.items())
```

Comment:
Perhaps we can be a bit more sophisticate here. If the length of the tensors are different sizes we also have a problem. So perhaps we first check that they are all the same size and then return true if that size is zero?

Reply:
In the case of attention masks, it makes sense that all tensors in the dictionary should have the same length. But in general, could there be a case when we might want to pass in multiple arguments of different sizes into a model?

Reply:
hmmm I suppose that's possible, but we're essentially checking on batch size here and I think we'd want there to be information per batch element (even if it's duplicated). Maybe that's more strict than we want, but for now I'd vote we enforce equality here and loosen it if we need to later on?

Reply:
Sounds good to me!